src/d0a70_len_4fe0.c

```diff
@@ -279,122 +314,175 @@ s16 func_8013A8E0(s32 idx) {
-    arg0->unk_02 = 0;
-    arg0->unk_08 = 0;
-    arg0->unk_0A = 0;
-    arg0->unk_0C = 0;
-    arg0->unk_0E = 0;
-    arg0->unk_1C[0][3] = 255;
-    arg0->unk_1C[1][3] = 255;
-    arg0->unk_03 = 0;
-    arg0->unk_04 = 0;
-    arg0->unk_08 = 0;
-    arg0->unk_0A = 0;
-
-    for (i = 0; i < ARRAY_COUNT(arg0->unk_1C); i++) {
-        for (j = 0; j < ARRAY_COUNT(arg0->unk_1C[0]); j++) {
-            arg0->unk_1C[i][j] = 0;
+    state->unk_10 = -1;
+    state->unk_05 = 0;
+    state->unk_06 = 0;
+    state->flags = 0;
+    state->meshType = 0;
+    state->renderType = 0;
+    state->firstVtxIdx = 0;
+    state->lastVtxIdx = 0;
+    state->unk_0C = 0;
+    state->unk_0E = 0;
+    state->unk_1C[0][3] = 255;
+    state->unk_1C[1][3] = 255;
+    state->subdivX = 0;
+    state->subdivY = 0;
+    state->firstVtxIdx = 0;
+    state->lastVtxIdx = 0;
+
+    for (i = 0; i < ARRAY_COUNT(state->unk_1C); i++) {
+        for (j = 0; j < ARRAY_COUNT(state->unk_1C[0]); j++) {
+            state->unk_1C[i][j] = 0;
         }
     }
 
-    for (i = 0; i < ARRAY_COUNT(arg0->unk_3C); i++) {
-        for (j = 0; j < ARRAY_COUNT(arg0->unk_3C[0]); j++) {
-            arg0->unk_3C[i][j] = 0;
+    for (i = 0; i < ARRAY_COUNT(state->unk_3C); i++) {
+        for (j = 0; j < ARRAY_COUNT(state->unk_3C[0]); j++) {
+            state->unk_3C[i][j] = 0;
         }
     }
 }
 
-INCLUDE_ASM(s32, "d0a70_len_4fe0", func_8013AA9C);
+INCLUDE_ASM(s32, "d0a70_len_4fe0", fold_update);
 
-void func_8013AF20(s32 idx, u16 arg1, s32 arg2) {
-    if ((*D_80156954)[idx].unk_14 & 1) {
-        if (arg2 != 0) {
-            (*D_80156954)[idx].unk_14 |= arg1;
+void fold_set_state_flags(s32 idx, u16 flagBits, s32 mode) {
+    if ((*D_80156954)[idx].flags & 1) {
+        if (mode) {
+            (*D_80156954)[idx].flags |= flagBits;
         } else {
-            (*D_80156954)[idx].unk_14 &= ~arg1;
+            (*D_80156954)[idx].flags &= ~flagBits;
         }
     }
 }
 
-INCLUDE_ASM(s32, "d0a70_len_4fe0", func_8013AF70);
+s32 fold_appendGfx_component(s32 idx, FoldImageRec* image, u32 flagBits, Matrix4f mtx) {
+    FoldState* state = &(*D_80156954)[idx];
+    s32 ret = 0;
+
+    if (image->unk_10 == 0) {
+        return 0;
+    }
+
+    state->arrayIdx = idx;
+    state->flags |= flagBits;
+    fold_currentImage->raster = image->raster;
+    fold_currentImage->palette = image->palette;
+    fold_currentImage->width = image->width;
+    fold_currentImage->height = image->height;
+    fold_currentImage->xOffset = image->xOffset;
+    fold_currentImage->yOffset =  image->yOffset;
+    fold_currentImage->unk_18 = 0;
+    fold_currentImage->unk_1E = 0;
+    fold_currentImage->gfxOtherModeD = image->unk_10;
+
+    if ((u32)idx >= 90) {
+        return 0;
+    }
 
-void func_8013B0EC(UnkD0A70Struct* arg0) {
-    switch (arg0->unk_01) {
+    if (idx >= 90 || state == NULL) {
+        return 0;
+    }
+
+    func_8013B0EC(state);
+    func_8013B1B0(state, mtx);
+
+    if (state->flags & 0x1000) {
```

Comment:
Seems like a potentially new flag enum for enums.h - Would be nice if you could add a dummy for it with unk flags